### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by blcts**
[DEMO LINK](https://blcts.github.io/react_goods-selector/)

### CODE REVIEW

**Review by alexbardyshev — APPROVED** [review]
Good job 😌✨

File: src/App.tsx [hunk 1]
```diff
@@ -1,51 +1,86 @@
...
+                  if (selectedGoods.includes(good)) {
+                    selectedGoods.splice(selectedGoods.indexOf(good), 1);
+                    this.setState({
+                      selectedGoods,
+                    });
+                  } else {
+                    selectedGoods.push(good);
+                    this.setState({
+                      selectedGoods,
+                    });
```
Comment (ivan-kadykalo) [inline src/App.tsx:73]: `? :`

File: src/App.tsx [hunk 2]
```diff
@@ -1,51 +1,86 @@
...
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title title">
+            { selectedGoods.length === 0
+              ? 'No goods selected'
+              : `${selectedGoods} is selected`}
```
Comment (ivan-kadykalo) [inline src/App.tsx:25]: Where is `are`?

**Review by ivan-kadykalo — CHANGES_REQUESTED** [review]
Add some spaces between rows, buttons are glued together
image

File: src/App.tsx [hunk 3]
```diff
@@ -1,51 +1,89 @@
...
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title title">
+            { selectedGoods.length === 0
+              ? 'No goods selected'
+              : (`${selectedGoods.join(', ')} ${selectedGoods.length === 1
+                ? 'is'
+                : 'are'
+              } selected`)}
```
Comment (Ihor-Karpyn) [inline src/App.tsx:28]: move to method

File: src/App.tsx [hunk 4]
```diff
@@ -1,51 +1,89 @@
...
+
+          {selectedGoods.length > 0 && (
+            <button
+              type="button"
+              className="
+                App__clear
+                button
+                is-small"
+              onClick={() => {
+                this.setState({ selectedGoods: [] });
```
Comment (Ihor-Karpyn) [inline src/App.tsx:39]: move to method

File: src/App.tsx [hunk 5]
```diff
@@ -1,51 +1,89 @@
...
+              Clear
+            </button>
+          )}
+        </header>
+
+        <ul className="container">
+          {goodsFromServer.map(good => (
+            <li className={classNames(
+              'Good',
+              selectedGoods.includes(good)
```
Comment (Ihor-Karpyn) [inline src/App.tsx:52]: `selectedGoods.includes(good)` don't repeat yourself, create const for it

File: src/App.tsx [hunk 6]
```diff
@@ -1,51 +1,89 @@
...
+            >
+
+              {good}
+
+              <button
+                type="button"
+                className={`Good__${selectedGoods.includes(good)
+                  ? 'remove is-danger is-outlined'
+                  : 'select is-success'}
+                  button is-small`}
```
Comment (Ihor-Karpyn) [inline src/App.tsx:65]: https://www.npmjs.com/package/classnames

File: src/App.tsx [hunk 7]
```diff
@@ -1,51 +1,89 @@
...
+                      && this.setState({
+                        selectedGoods,
+                      })
+                    )
+                    : (selectedGoods.push(good)
+                        && this.setState({
+                          selectedGoods,
+                        })
+                    )
+                )}
```
Comment (Ihor-Karpyn) [inline src/App.tsx:78]: splice and push is a mutable method, u can't mutate state, let's use the filter. And move to method

**Pull request closed without merging**